spec/requests/admin/site_settings_controller_spec.rb

```diff
@@ -727,8 +720,85 @@ def expect_user_count(
     end
 
     context "when logged in as a non-staff user" do
+      fab!(:user1, :user)
+      fab!(:user2, :user)
+
+      let(:tracking) { NotificationLevels.all[:tracking] }
+      let(:watching) { NotificationLevels.all[:watching] }
+
       before { sign_in(user) }
 
+      context "with default categories" do
+        let(:category_ids) { 3.times.collect { Fabricate(:category).id } }
+
+        before do
+          SiteSetting.default_categories_watching = category_ids.first(2).join("|")
+
+          CategoryUser.create!(
+            category_id: category_ids.last,
+            notification_level: tracking,
+            user: user2,
+          )
+
+          Jobs.run_immediately!
+        end
+
+        it "doesn't update default categories" do
+          put "/admin/site_settings/default_categories_watching.json",
+              params: {
+                default_categories_watching: category_ids.last(2).join("|"),
+                update_existing_user: true,
+              }
+
+          expect(response.status).to eq(404)
+
+          expect_not_enqueued_with(
+            job: Jobs::SiteSettingUpdateDefaultCategories,
+            args: {
+              id: "default_categories_watching",
+              value: category_ids.last(2).join("|"),
+              previous_value: category_ids.first(2).join("|"),
+            },
+          )
+
+          expect(
+            CategoryUser.where(category_id: category_ids.last, notification_level: watching).count,
+          ).to eq(0)
+        end
+      end
+
+      context "with default tags" do
+        let(:tags) { 3.times.collect { Fabricate(:tag) } }
```

Comment:
```suggestion
        fab!(:tag, :tag_1)
        fab!(:tag, :tag_2)
        fab!(:tag, :tag_3)
        let(:tags) { [tag_1, tag_2, tag_3] } 
```

Introduces the use of `fab!` so that we can reuse the fabricated objects between tests.